quic/s2n-quic/src/connection/handle.rs

```diff
@@ -169,6 +169,14 @@ macro_rules! impl_handle_api {
             self.0.application_protocol()
         }
 
+        /// Returns the application context the connection is using.
+        #[inline]
+        pub fn take_application_context(
```

Comment:
Thinking about the name a bit more: this is really the TLS context. We have quite a few other contexts in the library so it's probably good to clarify that detail.